sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -476,16 +476,21 @@ abstract class JdbcDialect extends Serializable with Logging {
   @Since("3.3.0")
   def isSupportedFunction(funcName: String): Boolean = false
 
+  /**
+   * The [[JDBCSQLBuilder]] that match database dialect.
+   */
+  @Since("4.0.0")
+  protected[jdbc] def jdbcSQLBuilder(): JDBCSQLBuilder = new JDBCSQLBuilder()
+
   /**
    * Converts V2 expression to String representing a SQL expression.
    * @param expr The V2 expression to be converted.
    * @return Converted value.
    */
   @Since("3.3.0")
   def compileExpression(expr: Expression): Option[String] = {
-    val jdbcSQLBuilder = new JDBCSQLBuilder()
```

Comment:
I'm not sure if it's worthwhile to add a new API for saving like copying 10 line of code...  but I don't have a strong preference on this.

cc @milastdbx